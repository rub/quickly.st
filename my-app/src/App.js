import React, { Component } from 'react';
import Header from './Header/Header.js';
import Nav from './Nav/Nav.js';
import ItemGenerator from './ItemGenerator/ItemGenerator.js';
import ItemsList from './ItemsList/ItemsList.js';
import Footer from './Footer/Footer.js';
import Help from './Help/Help.js';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>App</h1>
        <Header />
        <Nav />
        <Footer />
        <ItemGenerator />
        <ItemsList />
        <Help />
      </div>
    );
  }
}

export default App;
