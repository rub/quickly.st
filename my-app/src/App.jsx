import React from 'react';
import ItemGenerator from './ItemGenerator/ItemGenerator';
import ItemsList from './ItemsList/ItemsList';

const items = [
  {
    id: 1,
    task: 'This is the first item',
  },
  {
    id: 2,
    task: 'This is the second item',
  },
  {
    id: 3,
    task: 'This is the third item',
  },
];

class App extends React.Component {
  constructor(props) {
    super(props);

    // Set up the state
    this.state = {
      items,
      value: '',
    };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({ value: event.target.value });
  }

  render() {
    return (
      <div className="App">
        <ItemGenerator inputValue={this.state.value} onInputChange={this.handleChange} />
        <ItemsList items={this.state.items} />
      </div>
    );
  }
}

export default App;
