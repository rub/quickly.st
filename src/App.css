.app {
  background: linear-gradient(to bottom, #394C64, #212D3E);
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  padding: 6.854rem 1rem 0;
  width: 100%;
}

.app .app-container {
  max-width: 768px;
  width: 100%;
}
