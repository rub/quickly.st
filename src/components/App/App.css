.App {
  background: linear-gradient(to bottom, #394C64, #212D3E);
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  padding: 1rem;
  width: 100%;
}

.App-container {
  max-width: 768px;
  padding-top: 4.236rem;
  width: 100%;
}
