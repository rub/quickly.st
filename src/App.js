import React from 'react';
import TodoForm from './TodoForm';
import TodoList from './TodoList';

import './App.css';


class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      items: [],
      text: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleDelete = this.handleDelete.bind(this);
  }

  /**
   * Update the input text state while user is writing.
   */
  handleChange(e) {
    this.setState({
      text: e.target.value
    });
  }

  /**
   * Display a todo item based on input text value.
   * Clear the input text once user submitted the item.
   */
  handleSubmit(e) {
    e.preventDefault();

    if(!this.state.text) {
      return;
    }

    const newItem = {
      text: this.state.text,
      id: Date.now()
    }

    this.setState(prevState => ({
      items: prevState.items.concat(newItem),
      text: ''
    }))
  }

  /**
   * Create a new array that contains every item except the one user is
   * removing. To do this we use the filter method to check the key of the
   * clicked item against all of the items of the list.
   * @param  {string} key - The item's key user is removing.
   * @return {array} An array of remaining todo items.
   */
  handleDelete(key) {
    var filteredItems = this.state.items.filter(item => item.id !== key);

    this.setState({
      items: filteredItems
    })
  }

  render() {
    return (
      <div className="app">
        <div className="app-container">
          <TodoForm
            text={this.handleChange}
            submit={this.handleSubmit}
            inputValue={this.state.text}
          />
          <TodoList
            items={this.state.items}
            delete={this.handleDelete}
          />
        </div>
      </div>
    )
  }
}

export default App;
