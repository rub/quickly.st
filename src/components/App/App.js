import React from 'react';

import './App.css';

class App extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="App">
        <div className="App-container"></div>
      </div>
    )
  }
}

export default App;