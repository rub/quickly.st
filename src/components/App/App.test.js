import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';


it('renders the app without crashing', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})