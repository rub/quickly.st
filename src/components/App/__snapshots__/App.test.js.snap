// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app without crashing 1`] = `
<div
  className="App"
>
  <div
    className="App-container"
  />
</div>
`;
